firebase-case-study.md

```diff
@@ -0,0 +1,393 @@
+
+**Backups** - *The apps are protected from any possibility of data loss by relying on the automatic backup feature of this platform. Clicking the Backups tab of the Firebase Database gives one access to the auto-backup settings*.
+
+____________________________________________________________________________________________________________________
+
+## **Weaknesses:**
+
+![Weakneses](images/weaknesses.jpeg)
+
+- It’s not Open-Source
+- Firebase does not work in many countries
+- Only NoSQL databases are available
+- Slow Queries
+- Not all services are free to start
+- It’s not cheap and pricing is difficult to forecast
+- Runs only on Google Cloud
+- Dedicated Servers & Enterprise Support are not an option
+
+____________________________________________________________________________________________________________________
+
+**It’s not Open-Source** - *Firebase users can't modify Firebase source code because it is closed-source. Because of this issue, developers using Firebase end up using alternative platforms*.
+
+**Firebase does not work in many countries** - *Firebase is a subdomain from Google that only works in some countries*.
+
+**Only NoSQL databases are available** - *In Firebase, there is no option to use a relational database. There is a solution for this situation, and a user can craft a complex with Firebase. The implementation may be complex for a beginner using the platform. Firebase uses JSON, and there are virtually no SQL features, so migrating from the database will be more challenging*.
+
+**Slow Queries** - *Its specific limitations are a downfall of using it, like the frequency limited to 1 per second. No native aggregation queries and the maximum API request size is only 10 MiB. The document size limit is 1 MiB, and the Queries have slow performance*.
+
+**It's not free or cheap, and pricing is complex to forecast** - *Most of its services are available under the Spark Plan, but some are unavailable. Price changes once the user exceeds the free limit. Like the Cloud Vision APIs under the Machine Learning feature, it is only available to Blaze Plan users and charges $1.5/k APIs. Firebase is a feature- high-priced mobile application development platform. It offers a free plan, but that comes with some restrictions. In another way, some open-source technologies deliver more affordable solutions. Estimating Firebase's costs can be difficult for some users, especially when the scalability requirements are unpredictable. The pricing is complicated, and Firebase has specific pricing for every feature. Keeping the costs under a threshold can be challenging with the platform, and there is no easy way to cap or put hard limits on the bill. Understand how the billing works before committing to this model and monitor your costs regularly to avoid unexpected charges by the end of the month*.
```

Comment:
```suggestion
**It's not free or cheap, and pricing is complex to forecast** - *Most of its services are available under the Spark Plan, but some are unavailable. Price changes once the user exceeds the free limit. Like the Cloud Vision APIs under the Machine Learning feature, it is only available to Blaze Plan users and charges $1.5/k API requests. Firebase is a high-priced mobile application development platform. It offers a free plan, but that comes with some restrictions. In another way, some open-source technologies deliver more affordable solutions. 

Estimating Firebase's costs can be difficult for some users, especially when the scalability requirements are unpredictable. The pricing is complicated, and Firebase has specific pricing for every feature. Keeping the costs under a threshold can be challenging with the platform, and there is no easy way to cap or put hard limits on the bill. Understand how the billing works before committing to this model and monitor your costs regularly to avoid unexpected charges by the end of the month*.
```